py3status/modules/diskdata.py

```diff
@@ -0,0 +1,171 @@
+            # time from previous check
+            ios = self._get_io_stats(self.disk)
+            timedelta = time() - self.last_time
+
+            read = ios[0] - self.last_stat[0]
+            write = ios[1] - self.last_stat[1]
+
+            # update last_ info
+            self.last_stat = self._get_io_stats(self.disk)
+            self.last_time = time()
+
+            read /= timedelta
+            write /= timedelta
+
+            total = read + write
+
+            self.values['read'] = self._format(read)
+            self.values['total'] = self._format(total)
+            self.values['write'] = self._format(write)
+            self.py3.threshold_get_color(read, 'read')
+            self.py3.threshold_get_color(total, 'total')
+            self.py3.threshold_get_color(write, 'write')
+
+        if '{free}' in self.format or '{used' in self.format:
+            free, used, used_percent = self._get_free_space(self.disk)
+
+            self.values['free'] = self.format_space.format(free)
+            self.values['used'] = self.format_space.format(used)
+            self.values['used_percent'] = self.format_space.format(used_percent)
+            self.py3.threshold_get_color(free, 'free')
+            self.py3.threshold_get_color(used, 'used')
+
+        return {'cached_until': self.py3.time_in(self.cache_timeout),
+                'full_text': self.py3.safe_format(self.format, self.values)}
+
+    def _get_free_space(self, disk):
+        if not disk:
+            disk = '/'
+        elif not disk.startswith('/dev/'):
+            disk = '/dev/' + disk
+
+        total = 0
+        used = 0
+        free = 0
+
+        df = subprocess.check_output(['df']).decode('utf-8')
+        for line in df.split('\n'):
+            if line.startswith(disk):
+                data = line.split()
+                total += int(data[1]) / 1024 / 1024
+                used += int(data[2]) / 1024 / 1024
+                free += int(data[3]) / 1024 / 1024
+
+        if total == 0:
+            return free, used, 'err'
+
+        return free, used, 100 * used / total
+
+    def _get_io_stats(self, disk):
+        if disk and disk.startswith('/dev/'):
+            disk = disk[5:]
+        read = 0
+        write = 0
+        with open('/proc/diskstats', 'r') as fd:
+            for line in fd:
+                if disk and disk in line:
+                    data = line.split()
+                    read += int(data[5]) * self.sector_size
+                    write += int(data[9]) * self.sector_size
+                else:
+                    data = line.split()
+                    if data[1] == '0':
+                        read += int(data[5]) * self.sector_size
+                        write += int(data[9]) * self.sector_size
+        return read, write
+
+    def _format(self, value):
+        """
+        Return formatted string
+        """
+        value, unit = self.py3.format_units(value, unit="b/s", si=True)
```

Comment:
As before I'm not sure si is a good idea as people expect none si eg 2^10 units even if they write `MB` we can help educate.

Also is this actually bits/second I suspect it is `B/s`